src/components/ApplicationContext/ApplicationContext.tsx

```diff
@@ -127,8 +160,64 @@ export const ApplicationProvider: React.FC = () => {
       .catch(() => setStatus(AppStatus.Errored))
   }, [])
 
+  const savePage = () => {
```

Comment:
I changed this to save current page only, so we won't save unvalidated inputs in the other pages.